### PR TITLE
Drop DiGraph

### LINKED ISSUE #3136: [FEA]: Drop DiGraph
### Is this a new feature, an improvement, or a change to existing functionality?

Change

### How would you describe the priority of this feature request

Medium

### Please provide a clear description of problem this feature solves

The DiGraph class has been deprecated for over a year.   It is time to drop that code

### Describe your ideal solution

Removed DiGraph

### Describe any alternatives you have considered

_No response_

### Additional context

remove DiGraph

### Code of Conduct

- [X] I agree to follow cuGraph's Code of Conduct
- [X] I have searched the [open feature requests](https://github.com/rapidsai/cugraph/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22%2Cimprovement%2Cenhancement) and have found no duplicates for this feature request

### PR DESCRIPTION
**Pull request by BradReesWork**
this PR removes the DiGraph classes and all references to those classes from the code and tests.
All test are updated and validated.

DiGraph has been deprecated for over a year and it was time to finally remove it

closes #3136

### CODE REVIEW

File: python/cugraph/cugraph/components/connectivity.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def weakly_connected_components(G, directed=None, connection=None, return_labels
             For non-Graph-type (eg. sparse matrix) values of G only.
             Raises TypeError if used with a Graph object.
 
-        If True, then convert the input matrix to a cugraph.DiGraph
+        If True, then convert the input matrix to a Graph(ditected=True)
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/components/connectivity.py:124]:
```suggestion
        If True, then convert the input matrix to a Graph(directed=True)
```

File: python/cugraph/cugraph/tests/test_balanced_cut.py [hunk 2]
```diff
@@ -97,26 +97,24 @@ def test_edge_cut_clustering_with_edgevals(graph_file, partitions):
...
-
-    with pytest.deprecated_call():
-        G = cugraph.DiGraph()
-    G.from_cudf_edgelist(
-        df, source="src", destination="dst", edge_attr="val", renumber=False
-    )
-
-    with pytest.raises(Exception):
-        cugraph_call(G, 2)
+#def test_digraph_rejected():
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/tests/test_balanced_cut.py:102]: Should this be commented out?  Is the intended behavior of balanced cut to raise an exception for directed graphs?

File: python/cugraph/cugraph/tests/test_bfs.py [hunk 3]
```diff
@@ -405,7 +405,7 @@ def test_bfs(gpubenchmark, dataset_nxresults_startvertex_spc, cugraph_input_type
         elif cugraph_input_type is nx.Graph:
             cugraph_input_type = nx.DiGraph
 
-    if not isinstance(cugraph_input_type, (cugraph.Graph, cugraph.DiGraph)):
+    if not isinstance(cugraph_input_type, (cugraph.Graph)):
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/tests/test_bfs.py:408]:
```suggestion
    if not isinstance(cugraph_input_type, cugraph.Graph):
```

File: python/cugraph/cugraph/tests/test_connectivity.py [hunk 4]
```diff
@@ -306,7 +306,7 @@ def test_weak_cc(gpubenchmark, dataset_nxresults_weak, cugraph_input_type):
 
     # cuGraph or nx 'input_type' should have this parameter set to None
     directed = None
-    if not isinstance(cugraph_input_type, (cugraph.Graph, cugraph.DiGraph)):
+    if not isinstance(cugraph_input_type, (cugraph.Graph)):
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/tests/test_connectivity.py:309]:
```suggestion
    if not isinstance(cugraph_input_type, cugraph.Graph):
```

File: python/cugraph/cugraph/tests/test_connectivity.py [hunk 5]
```diff
@@ -370,7 +370,7 @@ def test_strong_cc(gpubenchmark, dataset_nxresults_strong, cugraph_input_type):
         api_type,
     ) = dataset_nxresults_strong
 
-    if not isinstance(cugraph_input_type, (cugraph.Graph, cugraph.DiGraph)):
+    if not isinstance(cugraph_input_type, (cugraph.Graph)):
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/tests/test_connectivity.py:373]:
```suggestion
    if not isinstance(cugraph_input_type, cugraph.Graph):
```

File: python/cugraph/cugraph/tests/test_louvain.py [hunk 6]
```diff
@@ -116,7 +116,7 @@ def test_louvain_csr_graph(is_weighted):
     karate = DATASETS_UNDIRECTED[0]
     df = karate.get_edgelist()
 
-    M = cupy.sparse.coo_matrix(
+    M = cupyx.scipy.sparse.coo_matrix(
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/tests/test_louvain.py:119]: What is the reason for this change?

File: python/cugraph/cugraph/tests/test_sssp.py [hunk 7]
```diff
@@ -223,7 +223,7 @@ def test_sssp(gpubenchmark, dataset_source_nxresults, cugraph_input_type):
     # Extract the params generated from the fixture
     (G, dataset_path, source, nx_paths, Gnx) = dataset_source_nxresults
 
-    if not isinstance(cugraph_input_type, (cugraph.Graph, cugraph.DiGraph)):
+    if not isinstance(cugraph_input_type, (cugraph.Graph)):
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/tests/test_sssp.py:226]:
```suggestion
    if not isinstance(cugraph_input_type, cugraph.Graph):
```

File: python/cugraph/cugraph/traversal/bfs.py [hunk 8]
```diff
@@ -43,7 +43,7 @@ def _ensure_args(G, start, i_start, directed):
 
     G_type = type(G)
     # Check for Graph-type inputs
-    if (G_type in [Graph, DiGraph]) or is_nx_graph_type(G_type):
+    if (G_type in [Graph]) or is_nx_graph_type(G_type):
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/traversal/bfs.py:46]:
```suggestion
    if G_type is Graph or is_nx_graph_type(G_type):
```

File: python/cugraph/cugraph/traversal/bfs.py [hunk 9]
```diff
@@ -93,7 +93,7 @@ def _convert_df_to_output_type(df, input_type):
     Given a cudf.DataFrame df, convert it to a new type appropriate for the
     graph algos in this module, based on input_type.
     """
-    if input_type in [Graph, DiGraph]:
+    if input_type in [Graph]:
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/traversal/bfs.py:96]:
```suggestion
    if input_type is Graph:
```

File: python/cugraph/cugraph/traversal/sssp.py [hunk 10]
```diff
@@ -46,7 +46,7 @@ def _ensure_args(
 
     G_type = type(G)
     # Check for Graph-type inputs
-    if (G_type in [Graph, DiGraph]) or is_nx_graph_type(G_type):
+    if (G_type in [Graph]) or is_nx_graph_type(G_type):
```
Comment (alexbarghi-nv) [inline python/cugraph/cugraph/traversal/sssp.py:49]:
```suggestion
    if G_type is Graph or is_nx_graph_type(G_type):
```

**Review by alexbarghi-nv — CHANGES_REQUESTED** [review]
looks good, just minor changes requested and one question

**Review by acostadon — APPROVED** [review]
Looks good.

File: python/cugraph/cugraph/sampling/node2vec.py [hunk 11]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2022, NVIDIA CORPORATION.
+# Copyright (c) 2023, NVIDIA CORPORATION.
```
Comment (acostadon) [inline python/cugraph/cugraph/sampling/node2vec.py:1]: should be 2022-2023
I think.

File: python/cugraph/cugraph/tests/test_balanced_cut.py (hunk 2 above)
Comment (BradReesWork) [reply python/cugraph/cugraph/tests/test_balanced_cut.py]: I dropped that code.  It was really only testing that an exception was raised when a DiGraph was created

File: python/cugraph/cugraph/tests/test_louvain.py (hunk 6 above)
Comment (BradReesWork) [reply python/cugraph/cugraph/tests/test_louvain.py:119]: "cupy.sparse.coo_matrix" is depricated

Comment (alexbarghi-nv) [reply python/cugraph/cugraph/tests/test_louvain.py:119]: ok

**Review by alexbarghi-nv — APPROVED** [review]
👍

File: python/cugraph/cugraph/sampling/node2vec.py (hunk 11 above)
Comment (BradReesWork) [reply python/cugraph/cugraph/sampling/node2vec.py:1]: thanks - I was in a flow of just updating the date and missed that this should have been a range

**Review by acostadon — APPROVED** [review]
Looks good to me

**Comment by BradReesWork** [comment]
rerun tests
